File: javascript_files/n-body.js
function main() {
    let total = 0;
    const n = 50000000;
    
    for(let i = 0; i < n; i++) {
        const a = i * 3.14159;
        const b = a / 2.71828;
        const c = a + b - i;
        
        const temp = Math.floor(c);
        const modulo = temp - Math.floor(temp / 1000) * 1000;
        total += modulo;
    }
    
    return total;
}

main();

File: r_files/n-body.R
main <- function() {
  total <- 0
  n <- 50000000
  
  for (i in 0:(n - 1)) {
    a <- i * 3.14159
    b <- a / 2.71828
    c <- a + b - i
    
    temp <- as.integer(c)
    modulo <- temp - (temp %/% 1000) * 1000
    total <- total + modulo
  }
  
  return(total)
}

main()

